### PR TITLE
[MusicGen] Fix audio channel attribute

### PR DESCRIPTION
**Pull request by sanchit-gandhi**
# What does this PR do?

`audio_channels` attribute is stored in the decoder sub-model config, not the overall top-level config. 

The logits test for MusicGen mono previously failed on the daily CI tests (cc @ydshieh): https://github.com/huggingface/transformers/actions/runs/6806630933/job/18508277638
It passes when we fix this bug.

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing!
